presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -924,6 +945,22 @@ public void dropTable(HiveIdentity identity, String databaseName, String tableNa
         }
     }
 
+    private static void deleteDir(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path, boolean recursive)
```

Comment:
we should also call this for dropPartition call - since that will also face the same problem.